lib/rucio/core/lifetime_exception.py

```diff
@@ -207,28 +232,42 @@ def __add_exception(dids, account, pattern, comments, expires_at, estimated_volu
 
 
 @transactional_session
-def update_exception(exception_id, state, session=None):
+def update_exception(
+        exception_id: str,
+        state: LifetimeExceptionsState,
+        session: 'Optional[Session]' = None
+) -> None:
     """
     Update exceptions state to Lifetime Model.
 
     :param exception_id:   The id of the exception
     :param state:          The states to filter
     :param session:        The database session in use.
     """
-    query = session.query(models.LifetimeExceptions).filter_by(id=exception_id)
-    try:
-        query.first()
-    except NoResultFound:
-        raise LifetimeExceptionNotFound
-
-    if state in [LifetimeExceptionsState.APPROVED, LifetimeExceptionsState.REJECTED]:
-        query.update({'state': state, 'updated_at': datetime.utcnow()}, synchronize_session=False)
-    else:
+    ALLOWED_STATES = (LifetimeExceptionsState.APPROVED, LifetimeExceptionsState.REJECTED)
+    if state not in ALLOWED_STATES:
         raise UnsupportedOperation
 
+    query = update(
+        models.LifetimeExceptions
+    ).where(
+        models.LifetimeExceptions.id == exception_id
+    ).values(
+        state=state,
+        updated_at=datetime.utcnow()
```

Comment:
`synchronize_session=False` was lost in translation

Reply:
`.execution_options(synchronize_session=False)`

Reply:
Previously there was a `SELECT` then an `UPDATE`. Is it still necessary?

Reply:
Yes, it is needed for the update. It's just to tell sqlalchemy to not update its python objects mapping to the database. I wonder why we don't dissabe this behavior all together. With so many queries using synchronize_session=False in our code, that mapping should be borderline useless anyway

Reply:
I’m afraid I do not understand. :(